pkg/charts/values_test.go

```diff
@@ -85,6 +85,10 @@ func TestNewValues(t *testing.T) {
 					Request: "",
 				},
 			},
+			Trace: &Trace{
+				CollectorSvcAddr:    "",
+				CollectorSvcAccount: "default",
```

Comment:
It looks like you're expecting the namespace further down, so this won't work?

Reply:
This test is to validate the defaults read from the `values.yaml` file. The logic that appends the namespace is in the `pkg/inject` package. The test in `pkg/inject/inject_test.go` will require the namespace suffix to be added as seen [here](https://github.com/linkerd/linkerd2/pull/3481/files#diff-61c5e7cb5fc7a736c4ba737d5ad5f4ebR142).

Reply:
I'm trying to understand this.  Does the injector use values.yaml as the defaults for the Values struct?  If so, I think this is a problem.  For any pods that don't have the annotation, these default values would be used and plugged into the proxy partial, leading to a `LINKERD2_PROXY_TRACE_COLLECTOR_SVC_NAME` of `default.serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)` which is missing the namespace.

Reply:
Wouldn't it be more appropriate for the Trace struct to be absent from Values.yaml?  So that the default value of `Trace` is nil?

Reply:
The injector doesn't use `values.yaml` during inject time. The `values.yaml` file is only relevant for `install`, when we work on https://github.com/linkerd/linkerd2/issues/3468. The reason for not defaulting `Trace` to nil here is because I'm hoping that this file is the only place where we will specify the default constant. 

> leading to a LINKERD2_PROXY_TRACE_COLLECTOR_SVC_NAME of default.serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain) which is missing the namespace.

The tracing environment variables won't be added if the `CollectorSvcAddr` is empty.

Reply:
This change to the `values.yaml` is a leftover from pre- a4b17184add0b3f6b9579a459e32e9ad3aea1bdc. Since I assume we will tackle #3468 very soon, I figure it's a small enough change to be left alone.

Reply:
I can confirm that when the annotation is absent, the environment variables are also absent.